### ISSUE TITLE
relative $ref incorrectly resolved from root rather than current URI (Introduced in 4.2.1)

### ISSUE THREAD

**Issue by dmccaffery** [mrin9/RapiDoc#17]
In 4.2.0, the following works for a open-api specification hosted in a sub path, such as https://docs.example.com/serviceA/

```yaml
components:
  responses:
    collection:
      content:
        application/json:
          schema:
            ...
          example:
            $ref: 'examples/collection.json'
```

The example is correctly loaded from: https://docs.example.com/serviceA/examples/collection.json

In 4.2.1, the relative $ref is incorrectly resolved as relative to root as:

https://docs.example.com/examples/collection.json

WORKAROUND:

Downgrade to 4.2.0

**Comment by mrin9** [comment]
if I understand correctly, 
you are saying the that the api-spec is hosted at `https://docs.example.com/serviceA`
and the `$ref:'examples/collection.json'` inside the spec is resolving to `https://docs.example.com/examples/collection.json`  ?

**Comment by dmccaffery** [comment]
The HTML file is also hosted under /serviceA. It is referencing an openAPI spec in the same path in this case.

**Comment by dmccaffery** [comment]
Also, the api spec is referenced as a relative URL.

**Comment by mrin9** [comment]
will it be possible for you to provide a test case ? like couple of partial specs

**Comment by mrin9** [comment]
@dmccaffery, Released 4.2.2 which should take care of the issue you reported, can you confirm if your case behaves the way you expect it to be ?
